plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/preferences/SSLCertificatesPreference.java

```diff
@@ -0,0 +1,210 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.preferences;
+
+import java.security.cert.X509Certificate;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+
+import org.eclipse.jface.preference.IPreferenceStore;
+import org.jboss.tools.openshift.common.core.utils.StringUtils;
+import org.jboss.tools.openshift.common.core.utils.X509CertificateParser;
+import org.jboss.tools.openshift.internal.ui.OpenShiftUIActivator;
+
+public class SSLCertificatesPreference {
+	final static String ALLOWED_CERTIFICATES = "allowed_certificates";
+	final static String SEPARATOR = ";";
+	final static String NL = StringUtils.getLineSeparator();
+
+	static SSLCertificatesPreference instance = new SSLCertificatesPreference();
```

Comment:
If it is approved, I can add the same for Openshift 2. 
Should it be just copied, as other functionality, or may I create abstract superclasses in openshift.common.ui to extend in Openshift 2 and Openshift 3?